### DISCUSSION TITLE
How to select all items from live_grep to quickfix ?

### DISCUSSION THREAD

**Discussion by lotusirous** [ibhagwan/fzf-lua#1211] [Q&A] [answered]
Follow up the discussion in https://github.com/ibhagwan/fzf-lua/discussions/1126 , 

> Another refinement options is to select all entries you’re interested in and press enter, by default multiple entries will go to quick fix, you can then go back and forth the quickfix wirh :cnext and :cprev (I map these to ]q and [q) and also explore these with :FzfLua quickfix.

How can I select all file in the result from `live_grep` or `grep` into quickfix list ?

There is a profile telesceope in : https://github.com/ibhagwan/fzf-lua/blob/a9e5755721c0f24cd1635ea4ff181df5b1bd5dd4/lua/fzf-lua/profiles/telescope.lua#L81
But it looks like there is no option for `grep`

Thanks

**Comment by ibhagwan** [comment]
Add this mapping to the default `keymap.fzf` and you’ll get the same effect for grep as well.

> There is a profile telesceope in

The profile is just a set of options sent to `setup` you can copy the contents of the profile (or only the parts you like) and configure the same options without using the profile directly.

**Comment by ibhagwan** [comment, marked as answer]
Also see the discussion in #1109 for how to use a custom function in conjunction with `select-all`.

**Comment by ibhagwan** [comment]
You can even do that in a command one-liner, try:
```lua
-- Can’t have an action set to ctrl-q or options will conflict
:FzfLua live_grep keymap.fzf.ctrl-q=select-all+accept actions.ctrl-q=false
```

**Reply by ibhagwan** [reply]
FYI, I've initiated https://github.com/junegunn/fzf/issues/3810 which is now merged into main, sending all to quickfix shuold be made considerably easier in the next release (>0.52.1).

**Comment by ibhagwan** [comment]
FYI, **once fzf `0.53` is released** and with the latest commit https://github.com/ibhagwan/fzf-lua/commit/e65b86532380c4e60490bc19b647d0c4faa58096, it is now simplified to use fzf prefixes with any fzf-lua action.

Using `select-all` with an exisiting action keybind (in this case `alt-q`) is as easy as:
```lua
-- This will work assuming `alt-q` is already defined in `actions`
:FzfLua live_grep keymap.fzf.alt-q=select-all
```

It is also possible to define this within the action itself (can be any action including custom):
```lua
:lua require("fzf-lua").files({
  actions ={
    -- "+accept" isn't required since `accept` is implied with actions
    ["ctrl-q"] = { fn = require"fzf-lua".actions.file_sel_to_qf, prefix = "select-all" }
  }
})
```

Obviously the above cab be specified during `setup` for permanency:
```lua
require("fzf-lua").setup({
  files = {
    actions = { ["ctrl-q"] = { fn = require"fzf-lua".actions.file_sel_to_qf, prefix = "select-all" } }
  }
})
```

**Reply by lotusirous** [reply]
Thanks a lot. This is a great improvement !!
